modules/javafx.controls/src/main/java/javafx/scene/control/skin/TableViewSkinBase.java

```diff
@@ -816,18 +832,24 @@ private void refreshView() {
         }
     }
 
-    // Handles the horizontal scrolling when the selection mode is cell-based
-    // and the newly selected cell belongs to a column which is not totally
-    // visible.
-    void scrollHorizontally() {
-        TableFocusModel<M,?> fm = getFocusModel();
+    /**
+     * Handles the horizontal scrolling when the selection mode is cell-based
+     * and the newly selected cell belongs to a column which is not totally
+     * visible.
+     */
+    public void scrollHorizontally() {
+        TableFocusModel<M, ?> fm = getFocusModel();
         if (fm == null) return;
 
         TC col = getFocusedCell().getTableColumn();
         scrollHorizontally(col);
     }
 
-    void scrollHorizontally(TC col) {
+    /**
+     * Programmatically scroll to the given column.
+     * @param col
+     */
+    protected void scrollHorizontally(TC col) {
```

Comment:
The scrollHorizontally() and scrollHorizontally(TC col) methods are called to respectively scroll to the selected column, or the given column.
This can be used at various places but some examples are directly linked to the methods above. If onSelectPreviousCell or onSelectNextCell is called, we may want to force the scroll to adapt in order to make the newly selected cell visible.